### PR TITLE
Cli server version

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andaaron — DISMISSED** [review]

**Review by peusebiu — DISMISSED** [review]

File: pkg/cli/client/server_info_cmd.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	zerr "zotregistry.io/zot/errors"
+	"zotregistry.io/zot/pkg/api/constants"
+)
+
+func NewServerInfoCommand() *cobra.Command {
+	serverInfoCmd := &cobra.Command{
+		Use:   "server-info",
```
Comment (rchincha) [inline pkg/cli/client/server_info_cmd.go:21]: Should we break this into subcmds instead?

zli registry info : return this info?
zli registry status : do a /v2 lookup?

Comment (andaaron) [reply pkg/cli/client/server_info_cmd.go:21]: What would be the zli output for the /v2 lookup?

Comment (rchincha) [reply pkg/cli/client/server_info_cmd.go:21]: `zli status <remote>`
does both /v2 lookup and returns server version

Comment (rchincha) [reply pkg/cli/client/server_info_cmd.go:21]: @laurentiuNiculae can you post an example output for this?

File: pkg/cli/client/server_info_cmd.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	}
+
+	_, err = makeGETRequest(ctx, mgmtEndpoint, username, password, config.VerifyTLS, config.Debug,
+		&serverInfo, config.ResultWriter)
+	switch err {
+	case nil:
+		serverInfo.Status = StatusOnline
+	case zerr.ErrPageNotFound:
+		serverInfo.Status = StatusOnline
+		serverInfo.ErrorMsg = "MGMT endpoint is not enabled"
```
Comment (andaaron) [inline pkg/cli/client/server_info_cmd.go:71]:
```suggestion
		serverInfo.ErrorMsg = fmt.sprintf("%s%s endpoint is not available", constants.RoutePrefix, const.ExtMgmt)
```

File: pkg/cli/client/server_info_cmd.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	serverInfoCmd.PersistentFlags().StringP(ConfigFlag, "c", "",
+		"Specify the registry configuration to use for connection")
+	serverInfoCmd.PersistentFlags().StringP(UserFlag, "u", "",
+		`User Credentials of zot server in "username:password" format`)
+	serverInfoCmd.Flags().StringP(OutputFormatFlag, "f", "text", "Specify the output format [text|json|yaml]")
+
+	return serverInfoCmd
+}
+
+const (
```
Comment (andaaron) [inline pkg/cli/client/server_info_cmd.go:47]: Why not move them at the top of the file?

File: pkg/cli/client/server_info_cmd.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+	mgmtEndpoint, err := combineServerAndEndpointURL(config.ServURL, fmt.Sprintf("%s%s",
+		constants.RoutePrefix, constants.ExtMgmt))
+	if err != nil {
+		return err
+	}
+
+	_, err = makeGETRequest(ctx, mgmtEndpoint, username, password, config.VerifyTLS, config.Debug,
+		&serverInfo, config.ResultWriter)
+	switch true {
```
Comment (andaaron) [inline pkg/cli/client/server_info_cmd.go:66]:
```suggestion
	switch {
```

File: pkg/cli/client/server_info_cmd.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+	case err == nil:
+		serverInfo.Status = StatusOnline
+	case errors.Is(err, zerr.ErrPageNotFound):
+		serverInfo.Status = StatusOnline
```
Comment (andaaron) [inline pkg/cli/client/server_info_cmd.go:70]: I'm not convinced about this. How do we know this is the correct url of a zot server? Maybe there's another app running there.

Comment (laurentiuNiculae) [reply pkg/cli/client/server_info_cmd.go:70]: How can we know for sure it's a zot server?

Comment (andaaron) [reply pkg/cli/client/server_info_cmd.go:70]: I think we should https://github.com/opencontainers/distribution-spec/blob/main/spec.md#determining-support

Comment (rchincha) [reply pkg/cli/client/server_info_cmd.go:98]: This is a liveness check, and not a "are you truly a zot or not" check

Comment (andaaron) [reply pkg/cli/client/server_info_cmd.go:98]: I was thinking the user may have misconfigured the server url.

**Comment by laurentiuNiculae** [comment]
Here are some examples of outputs 

```console
$ zli-linux-amd64 status --url http://127.0.0.1:5000 --user laur:password
Server Status: online
Server Version: bsd-0.0.4
Dist Spec Version: 1.1.0-dev
Built with: imagetrust, lint, metrics, mgmt, profile, scrub, search, sync, ui, userprefs
```
Bad Credentials:
```console
$ zli-linux-amd64 status --url http://127.0.0.1:5000 --user laur:bad-pass
Server Status: unknown
Error: unauthorised access, given credentials are invalid
```
No Credentials:
```console
$ zli-linux-amd64 status --url http://127.0.0.1:5000
Server Status: unknown
Error: unauthorised access, endpoint requires valid user credentials (add the flag '--user [user]:[password]')
```
MGMT is not enabled (minimal version):
```console
$ zli-linux-amd64 status --url http://127.0.0.1:5000
Server Status: online
Error: /v2/_zot/ext/mgmt endpoint is not available
```
Server is offline:
```console
$ zli-linux-amd64 status --url http://127.0.0.1:5000
Server Status: offline
Error: Get "http://127.0.0.1:8080/v2/": dial tcp 127.0.0.1:8080: connect: connection refused```
Given server isn't a OCI Registry:
```

**Review by andaaron — APPROVED** [review]

**Pull request merged**